src/main/java/hw3/components/LogSection.java

```diff
@@ -0,0 +1,55 @@
+package hw3.components;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import java.util.List;
+
+public class LogSection extends AbstractPageComposite {
+
+    @FindBy(css = ".logs li")
+    private List<WebElement> logs;
+
+    public LogSection(WebDriver driver) {
+        super(driver);
+    }
+
+    public boolean isCheckboxLogDisplayed(String checkboxName, String status) {
+        waitForLogsToBeVisible();
+        boolean isDisplayed = false;
+        for (WebElement log : logs) {
+            if (log.getText().contains(checkboxName + ": condition changed to " + status)) {
+                isDisplayed = true;
+            }
+        }
+        return isDisplayed;
+    }
+
+
+    public boolean isRadioLogDisplayed(String radioName) {
+        waitForLogsToBeVisible();
+        boolean isDisplayed = false;
+        for (WebElement log : logs) {
+            if (log.getText().contains("metal: value changed to " + radioName)) {
+                isDisplayed = true;
+            }
+        }
+        return isDisplayed;
+    }
+
+    public boolean isDropdownLogDisplayed(String dropdownValue) {
+        waitForLogsToBeVisible();
+        boolean isDisplayed = false;
+        for (WebElement log : logs) {
+            if (log.getText().contains("Colors: value changed to " + dropdownValue)) {
+                isDisplayed = true;
+            }
+        }
+        return isDisplayed;
+    }
```

Comment:
If you are looking to these methods you can see that here is code duplication.

Reply:
thanks, I ll fix this